### PR TITLE
Inhibit clicks on summary's children

### PR DESCRIPTION
**Pull request by jsha**
A byproduct of using `<details>` and `<summary>` to show/hide detailed documentation was that clicking any part of a method heading (or impl heading) would show or hide the documentation. This was not super noticeable because clicking a link inside the method heading would navigate to that link. But clicking any unlinked black text in a method heading would trigger the behavior.

That behavior was somewhat unexpected, and means that if you try to click a type name in a method heading, but miss by a few pixels, you get a confusing surprise.

This change inhibits that behavior by putting an event listener on most summaries that cancels the event unless the event target was the summary itself. In practice, that means it cancels the event unless the target was the "[+]" / "[-]", because the rest of the heading is wrapped inside a `<div>`, which is the target for anything that doesn't have a more specific target.

r? @Manishearth

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Please add a GUI test, otherwise it seems like a good idea. It often happens that when I highlight something, it closes the the doc, which is a bit annoying.

File: src/test/rustdoc-gui/toggle-click-deadspace.goml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// This test ensures that clicking on a method summary, but not on the "[-]",
+// doesn't toggle the <details>.
+debug: true
+goto: file://|DOC_PATH|/test_docs/struct.Foo.html
+size: (1000, 1000)
+assert-attribute: (".impl-items .rustdoc-toggle", {"open": ""})
+click: (249, 509) // This is the position of "pub" in "pub fn a_method"
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/toggle-click-deadspace.goml:7]: You can click directly on `h4.code-header`. I'd much prefer avoiding to use hard coded positions if possible. :)

File: src/test/rustdoc-gui/toggle-click-deadspace.goml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+assert-attribute: (".impl-items .rustdoc-toggle", {"open": ""})
+click: (229, 509) // This is the position of "[-]" next to that pub fn.
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/toggle-click-deadspace.goml:9]: Same here.

File: src/test/rustdoc-gui/toggle-click-deadspace.goml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// This test ensures that clicking on a method summary, but not on the "[-]",
+// doesn't toggle the <details>.
+debug: true
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/toggle-click-deadspace.goml:3]: Once you're done, please remove this one. ;)

**Comment by jsha** [comment]
I wrote a GUI test, but I couldn't get it to work. I tried to find the coordinates on which to simulate a click by loading the page in a regular browser, but when I tell the tester to click on the coordinates that correspond to the toggle, it doesn't appear to do anything. I tried using `--no-headless` but the browser window appears and disappears instantly. Any ideas how to debug?

**Comment by GuillaumeGomez** [comment]
Ah right, for that I use `wait-for: 600000` (for 10 min). It allows to keep the windows around to debug in peace.

File: src/test/rustdoc-gui/toggle-click-deadspace.goml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// This test ensures that clicking on a method summary, but not on the "[-]",
+// doesn't toggle the <details>.
+goto: file://|DOC_PATH|/test_docs/struct.Foo.html
+size: (1000, 1000)
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/toggle-click-deadspace.goml:4]: Do you still need this btw?

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me, one nit remaining but can be ignored.

File: src/test/rustdoc-gui/toggle-click-deadspace.goml (hunk 4 above)
Comment (jsha) [reply src/test/rustdoc-gui/toggle-click-deadspace.goml:4]: Nope! Removed.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Why the try?

**Comment by jsha** [comment]
Perhaps I misread the bors docs? Was attempting to trigger a normal re-run of the tests, since I can't reproduce the failure locally. And the CI status reported here doesn't show a failure, just a cancelled check? https://bors.tech/documentation/

**Comment by GuillaumeGomez** [comment]
It's easier this way. ;)

**Review by Manishearth — APPROVED** [review]

**Pull request merged**